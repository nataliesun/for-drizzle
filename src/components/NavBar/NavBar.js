import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class NavBar extends Component {

  state = {
    prevScrollpos: window.pageYOffset,
    visible: true
  }


  componentDidMount() {

    window.addEventListener("scroll", this.handleScroll);
  }

  handleLogoutClick = () => {

  };

  renderLogoutLink() {
    return (
      <>
        <li>
          <NavLink to='/dashboard' >
            Dashboard
          </NavLink>
        </li>
        <li>
          <NavLink to="/addProperty">
            <FontAwesomeIcon icon="plus" /> Property
          </NavLink>
        </li>
        <li>
          <NavLink to='/friends' >
            <FontAwesomeIcon icon="search" />{" "}
            Friends
          </NavLink>
        </li>
        <li onClick={this.handleLogoutClick}>
          Logout
        </li>
      </>
    );
  }

  renderLoginLink() {
    return (
      <>
        <li>
          <NavLink to="/register">
            Register
          </NavLink>
        </li>
        <li>
          <NavLink to='/login' >
            Login
          </NavLink>
        </li>
      </>
    );
  }

  navSlide = (ev) => {
    const burger = document.querySelector('.burger');
    const nav = document.querySelector('.nav-links');
    const navLinks = document.querySelectorAll('.nav-links li');


    nav.classList.toggle('nav-active');

    navLinks.forEach((link, index) => {
      if (link.style.animation) {
        link.style.animation = '';
      } else {
        link.style.animation = `navLinkFade 0.5s ease forwards ${index / 7 + 0.3}s`;
      }
    });

    //burger animation
    burger.classList.toggle('toggle');

  }

  handleScroll = () => {
    const { prevScrollpos } = this.state;

    const currentScrollPos = window.pageYOffset;
    const visible = prevScrollpos > currentScrollPos;

    this.setState({
      prevScrollpos: currentScrollPos,
      visible
    });
  };



  componentWillUnmount() {
    window.removeEventListener("scroll", this.handleScroll);
  }

  render() {
    const navClass = this.state.visible && this.state.prevScrollpos > 1 ? "NavBar back" : this.state.visible ? "NavBar" : "NavBar NavBar__hidden";
    const links = this.context.loggedIn ? this.renderLogoutLink() : this.renderLoginLink();
    return (
      <>

        <nav className={navClass}>
          <NavLink to="/">
            <FontAwesomeIcon icon="door-open" />
            <h1>
              For Drizzle
            </h1>
          </NavLink>
          <div className="burger" onClick={e => this.navSlide(e)}>
            <div className="line1"></div>
            <div className="line2"></div>
            <div className="line3"></div>
          </div>
          <ul className="nav-links">
            {links}
          </ul>

        </nav>
      </>
    );
  }
}

export default NavBar;