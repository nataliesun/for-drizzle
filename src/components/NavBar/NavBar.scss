@import '../../variables.scss';

.NavBar__hidden {
  top: -50px !important;
}


.back {
  // background: linear-gradient(159deg,#512dcf 31%,#10a3d5 100%) #512dcf;
  background: $main-gradient;
  padding-bottom: 20px;
  border-radius: 40px;
  margin: 20px 0;

  h1, li, a {
    color: $text-color !important;
  }
  .burger div {
    background: $text-color !important;
  }
}

.NavBar {
  z-index: 1000;
  position: fixed;
  top: 0;
  padding-top: 20px;
  transition: top 0.6s;
  padding-left: 20px;
  width: 100vw;
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: $app-max-width;
  left: 50%;
  transform: translateX(-50%);

  h1 {
    display: inline;
    padding-left: 10px;
  }

  a {
    color: $grey-color;
    text-decoration: none;
  }

  ul {
    list-style-type: none;
  }

  li {
    float: left;
    margin-right: 15px;
    // font-family: 'Barlow Condensed', sans-serif;
    font-size: 16px;
    // letter-spacing: 1px;
    position: relative;
    cursor: pointer;
    color: $text-color;
  }

  li::after {
    background-color: rgb(40, 93, 139);
    display: block;
    content: '';
    width: 0;
    height: 3px;
    border-radius: 4px;
    left: 0;
    bottom: -10px;
    transition: width 0.35s ease 0s;
    position: absolute;
  }

  li:hover::after {
    width: 100%;
  }

  .burger {
    padding-right: 20px;
    display: none;
    cursor: pointer;
  }

.burger div {
    width: 25px;
    height: 3px;
    background-color: $grey-color;
    margin: 5px;
    border-radius: 40px;
  }


  @media screen and (max-width: 768px) {
  .nav-links {
    display: flex;
    position: absolute;
    right: 0px;
    height: 100vh;
    top: 0;
    background: $main-gradient;


    border-top-left-radius: 40px;
    border-bottom-left-radius: 40px;
    flex-direction: column;
    width: 50%;
    transform: translateX(100%);
    transition: transform 0.5s ease-in;
    justify-content: space-evenly;
    z-index: 10;
    align-items: center;
  }

  .nav-links li {
    opacity: 0;
  }

  .burger {
    display: block;
    z-index: 11;
  }
}

.nav-active {
  transform: translateX(0%);
      -webkit-box-shadow: -22px 17px 33px 0px rgba(0,0,0,0.61);
-moz-box-shadow: -22px 17px 33px 0px rgba(0,0,0,0.61);
box-shadow: -22px 17px 33px 0px rgba(0,0,0,0.61);

   li a {
     color: $text-color;
   }
}

@keyframes navLinkFade {
  from {
    opacity: 0;
    transform: translateX(50px);
  }

  to {
    opacity: 1;
    transform: translateX(0px);
  }
}

.toggle .line1 {
  transform: rotate(-45deg) translate(-5px, 6px);
}

.toggle .line2 {
  opacity: 0;
}

.toggle .line3 {
  transform: rotate(45deg) translate(-5px, -6px);
}


nav {
  -webkit-transition: all .3s;
  transition: all .3s;
}
}

