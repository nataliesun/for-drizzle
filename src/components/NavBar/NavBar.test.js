import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import NavBar from './NavBar';
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faDoorOpen, faPlusCircle, faPlus, faTimes
} from '@fortawesome/free-solid-svg-icons'

library.add(
  faDoorOpen, // logo
  faPlusCircle,
  faPlus,
  faTimes
)

describe('NavBar route component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})