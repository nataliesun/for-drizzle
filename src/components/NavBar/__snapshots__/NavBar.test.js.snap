// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar route component renders the UI as expected 1`] = `
<div
  className="NavBar__outer"
>
  <nav
    className="NavBar"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-door-open fa-w-20 blue"
        data-icon="door-open"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M624 448h-80V113.45C544 86.19 522.47 64 496 64H384v64h96v384h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM312.24 1.01l-192 49.74C105.99 54.44 96 67.7 96 82.92V448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h336V33.18c0-21.58-19.56-37.41-39.76-32.17zM264 288c-13.25 0-24-14.33-24-32s10.75-32 24-32 24 14.33 24 32-10.75 32-24 32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <h1>
        EstateCloud
      </h1>
    </a>
    <div
      className="burger"
      onClick={[Function]}
    >
      <div
        className="line1"
      />
      <div
        className="line2"
      />
      <div
        className="line3"
      />
    </div>
    <ul
      className="nav-links"
    >
      <li>
        <a
          aria-current={null}
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
